Makefile-az.mk

```diff
@@ -232,8 +232,8 @@ az-rmnodeclaims-fin: ## Remove Karpenter finalizer from all nodeclaims (use with
 az-rmnodeclaims: ## kubectl delete all nodeclaims; don't wait for finalizers (use with care!)
 	kubectl delete --wait=false nodeclaims --all
 
-az-taintnodes: ## Run e2etests
-	kubectl taint nodes CriticalAddonsOnly=true:NoSchedule --all --overwrite
+az-taintnodes: ## Taint all system nodepool nodes
+	kubectl taint nodes CriticalAddonsOnly=true:NoSchedule --selector='kubernetes.azure.com/mode=system' --overwrite
```

Comment:
I would prefer to taint all nodes not just system

Reply:
I agree with Alex that it should not taint the karpenter nodes. Could look at tainting any "aks nodepool" nodes. Is there a particular reason you want all nodes?

I know for myself I'd prefer otherwise as I ended up accidentally tainting some karpenter nodes before

Reply:
Talked offline for a bit, around the fact that sometimes with stuck nodes being able to blanket taint them all and run tests on top can be useful. I think the scenario of just wanting system nodes will be more common. However, open to adding a taint all if it would be useful within the future.